joomla/joomla-cms
# Standardized the update route for extension installs

Continuing the `JInstallerAdapter` implementations and standardizing the adapter paths, this PR focuses in on the `update` path.

This PR mainly alters the logic for the component updates; every other adapter modified had already used this standard path.  Again, the language adapter is excluded and the library adapter is not modified either because of its approach to updates right now.

Also fixed with this PR is a regression in the uninstall route where the uninstall queries were not being run.
### Testing Instructions

The path now always follows the install route (which was being done for all extensions but components) and the internal code determines which actions to perform based on the active route.  Since some of the shared component methods are updated to accommodate for the update route, you'll want to install extensions first.  To test updates, you'll want to use a release that isn't the latest one for optimal testing.  After confirming everything still installs OK, you'll want to update the extension now, and this should be done using any regular update method.  The extensions should update correctly.

Now try to uninstall your test extensions.  If they have uninstall queries (many components do), ensure those queries get executed.
